editor/project_settings_editor.cpp

```diff
@@ -484,4 +479,14 @@ ProjectSettingsEditor::ProjectSettingsEditor(EditorData *p_data) {
 
 	get_ok_button()->set_text(TTR("Close"));
 	set_hide_on_ok(true);
+
+	inspector->set_restrict_to_basic_settings(true);
```

Comment:
Should probably fetch the `use_advanced_mode` setting first and only update the inspector based on that value. This will update the inspector twice if the setting was set.